internal/handlers/common/msg_saslstart.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgSASLStart is a common implementation of the SASLStart command.
+func MsgSASLStart(_ context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
```

Comment:
I look at the other files in the `common` dir, and I'm not sure if this one should be called `MsgSASLStart` or just `SASLStart`. Fore example, take a look at `validate.go`, I think there was some other naming idea behind it.

Reply:
It should be `MsgSASLStart`, as it is a full implementation. `Validate` is helper that has a different signature (it accepts a logger)

Reply:
@AlekSi ok, got it!

Reply:
`Validate` seems to be an exception. There are plenty of other messages called `MsgSomething`, in particular `MsgDebugError` or `MsgWhatsMyURI`.